databases/week3/homework_week3.sql

```diff
@@ -0,0 +1,134 @@
+CREATE database meal_sharing;
+use meal_sharing;
+CREATE TABLE meal(
+	`id` INT UNSIGNED AUTO_INCREMENT,
+    `title` VARCHAR(255) NOT NULL,
+    `description` TEXT(255) NOT NULL,
+    `location` VARCHAR(255) NOT NULL,
+    `when` DATETIME NOT NULL,
+    `max_reservations` INT UNSIGNED NOT NULL,
+    `price` DECIMAL(10,2) NOT NULL,
+    `created_date` DATE NOT NULL,
+    PRIMARY KEY (id)
+) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+
+CREATE TABLE reservation (
+    `id` INT UNSIGNED AUTO_INCREMENT,
+    `number_of_guests` INT UNSIGNED NOT NULL,
+    `meal_id` INT UNSIGNED NOT NULL,
+    `created_date` DATE NOT NULL,
+    `contact_phonenumber` VARCHAR(255) NOT NULL,
+    `contact_name` VARCHAR(255) NOT NULL,
+    `contact_email` VARCHAR(255) NOT NULL,
+    PRIMARY KEY (id),
+    CONSTRAINT `fk_reservation_meal` FOREIGN KEY (`meal_id`) REFERENCES `Meal` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+    ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+    
+    CREATE TABLE review (
+    `id` INT UNSIGNED AUTO_INCREMENT,
+    `title` VARCHAR(255) NOT NULL,
+    `description` TEXT(255) NOT NULL,
+    `meal_id` INT UNSIGNED NOT NULL,
+    `stars` INT UNSIGNED NOT NULL,
+    `created_date` DATE NOT NULL,
+    PRIMARY KEY (id),
+   CONSTRAINT `fk_review_meal` FOREIGN KEY (`meal_id`) REFERENCES `Meal` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+
+-- Queries for meal table 
+INSERT INTO meal (`title`,`description`,`location`,`when`,`max_reservations`,`price`,`created_date`)
+VALUES ('pizza sandwish', 'it has beef and cheese with dressing', 'vejle', '2022-08-31 12:30:00', 10, 1200, '2022-08-30'),
+        ('burger', 'chicken cheese burger with extra cheese', 'copenhagen', '2022-09-01 20:00:00', 5, 870, '2022-08-31'),
+        ('apple pie', 'pie desert made with pie', 'Aarhus', '2022-05-20 10:20:00', 20, 150, '2022-04-02' );
+-- get all meals
+select * 
+from meal; 
+-- get a meal from id=2
+select * 
+from meal 
+where id =2;
+-- update
+UPDATE meal 
+SET `max_reservations` =7,`price`=1200
+WHERE id =2;
+-- Delete
+DELETE FROM meal
+WHERE id = 3;
+
+-- Queries for reservation table 
+INSERT INTO `reservation`(`number_of_guests`,`meal_id`,`created_date`,`contact_phonenumber`,`contact_name`,`contact_email`)
+VALUES (4, 1, '2022-07-26', '60545755', 'Hanaa', 'hanaa@gmail.com'),
+        (3, 1, '2022-04-08', '12390293', 'Asma', 'asma@gmail.com'),
+        (8, 2, '2022-05-01', '58584608', 'Huda', 'huda@gmail.com');
```

Comment:
Nice, inserting multiple records at once. But be careful, if you get errors it will be difficult to find which values made the errors.